### PR TITLE
72 order model specs

### LINKED ISSUE #72: Order model specs

### PR DESCRIPTION
**Pull request by applegrain**
- add order item model tests
- add order tests
- break out orders controller #create into smaller methods

closes #72

### CODE REVIEW

File: spec/models/order_spec.rb [hunk 1]
```diff
@@ -1,4 +1,23 @@
 require 'rails_helper'
 
 RSpec.describe Order, type: :model do
+  let!(:user) { Fabricate(:user) }
+  let!(:item) { Fabricate(:item) }
+  let!(:status) { Status.create(name: "ordered") }
+  let!(:order) { Order.create(user_id: user.id, status_id: status.id) }
+  let!(:order_item) { OrderItem.create(order_id: order.id, item_id: item.id, quantity: 3) }
```
Comment (mrjaimisra) [inline spec/models/order_spec.rb]: nice using fabricate. i may end up modifying some of them as I add more validations.

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -50,4 +43,21 @@ def send_text_message
...
+
+  def empty_cart
+    session[:cart] = {}
+    cart.empty
+  end
+
+  def notify_user
+    send_text_message if current_user.phone_number
+    flash[:success] = "Order placed! Dinners on the way!"
+  end
```
Comment (mrjaimisra) [inline app/controllers/orders_controller.rb]: "if you love something, never put in the view" -Josh Cheek

File: spec/features/admin_add_item_spec.rb [hunk 3]
```diff
@@ -1,11 +1,10 @@
 require "rails_helper"
 
 RSpec.describe "an admin on their dashboards" do
-  let!(:user) { Fabricate(:user) }
+  let!(:user) { Fabricate(:user, role: 1) }
```
Comment (mrjaimisra) [inline spec/features/admin_add_item_spec.rb]: makes sense

Comment (applegrain) [inline spec/features/admin_add_item_spec.rb]: Yeah, me and Jason figured out on the bus yesterday that with fabricator you can override the defaults like that ^^

**Pull request merged**